### DISCUSSION TITLE
Animations and Storyboards in WebAssembly

### DISCUSSION THREAD

**Discussion by avenditti70** [unoplatform/uno#8583] [Q&A]
I tried some XAML storyboard animations in a WASM project created with Uno Platform. I was not able to control the animation from code-behind using .Begin() method of Sotryboard class. It worked in UWP but not in WASM.
Visual Studio intellisense showed that method as "not available" in WASM: is it the case?  I wasn't able to find documentation about limitations of XAML Animation support on WASM projects.
Could you please point me out some documentation about that (specifically for XAML Animations in WASM) and  / or some working sample code?
![image](https://user-images.githubusercontent.com/7794955/164389255-34a3055c-06d7-413d-b4a7-0ed4502560e6.png)

**Comment by jeromelaban** [comment]
Thanks for the report. This could be incorrect reporting of the availability of `Storyboard.Begin()`, as it is implemented:
https://github.com/unoplatform/uno/blob/a926d5d8e1e678df41fff7a76a980f6952901455/src/Uno.UI/UI/Xaml/Media/Animation/Storyboard.cs#L92-L112

This is what I'm seeing:
![image](https://user-images.githubusercontent.com/5839577/164455070-4a30693e-b7e6-4ea1-ba18-e144a07e073d.png)

Could you tell which version of visual studio you are running? Also, it seems that Intellisense was still `Starting`. Did the warning still appear after that message went away?

**Comment by avenditti70** [comment]
Thanks Jérôme.

I'm using VS2022 64bit (current version 17.1.4)

A part from Intellisense's hints, this was the behavior I experienced: the project was targeted both for WASM and for UWP, and Storyboard's Begin() method worked for UWP and not for WASM (animation didn't start, as if method was simply ignored, without errors or exceptions). Maybe I'm using an "old" version of Uno Platform framework libraries? Could you please advise about how to be sure I'm using the latest version?

________________________________

**Reply by jeromelaban** [reply]
Thanks. Can you provide a repro sample of what you're trying to do and attach it as a zip file here?

**Reply by jeromelaban** [reply]
Thanks for the repro. Here's the issue to track: https://github.com/unoplatform/uno/issues/8613.

**Reply by jeromelaban** [reply]
I've posted a workaround in https://github.com/unoplatform/uno/issues/8613.

**Comment by avenditti70** [comment]
Hi Jerome.
Thank you so much for your support.
Here you can find a zip with my simple test project:
https://1drv.ms/u/s!Aq7SJzwziiEvg0wbZAp2PqVp4ZaS?e=NiAczK
It contains just the head for UWP (to show that it is working) and for WASM (to show that it is not working).
The test is very simple (it was probably the first test I did in Silverlight 10 years ago): I have a MainPage.xaml with a Button that starts a Storyboard animation (hosted in the MainPage itself as a resource) for a "Bouncing Ball" (implemented as a rounded ellipse drawn on a canvas); in UWP the Storyboard correctly starts, in WASM it doesn't start at all (but I verified that the Click event is fired and executed, but the Storyboard.Begin seems to be ignored).

Also, Intellisense is now giving the opposite information I saw in my first post: Begin available for WASM, not available for UWP  (but the test showed something that could be the opposite...)

[cid:9ad87796-41e7-4ce5-baac-a3ac2b99e91c]

Maybe I'm missing something in terms of updates? I'm using VS2022 with the latest UnoPlatform solution templates.

Thank you,
Alberto


________________________________

**Reply by jeromelaban** [reply]
(make sure to stay on the opened conversation above, it's easier to follow)

**Comment by avenditti70** [comment]
Thank you very much, Jerome.
I was totally unaware of the EnableDependentAnimation property.
As far as I understand, this means that my animation was considered "resource intensive" on WASM and kept frozen in order to avoid unresponsive app.
In a certain way, this makes sense, even if I cant understand why on the same hardware the same animation is not frozen for performance reasons in WPF...
Maybe the WASM runtime is "considered less performant" than the WPF one... 🙂
Thanks again,
AV

________________________________

**Reply by jeromelaban** [reply]
This is actually an issue on Uno's side, where this should still work. We'll make changes for this to work without having to fiddle with `EnableDependentAnimation`.
